engine/runtime-instrument-id-execution/src/main/java/org/enso/interpreter/instrument/IdExecutionInstrument.java

```diff
@@ -250,7 +251,7 @@ public void onReturnExceptional(VirtualFrame frame, Throwable exception) {
     }
 
     private void onExpressionReturn(Object result, Node node, EventContext context) throws ThreadDeath {
-      boolean isPanic = result instanceof AbstractTruffleException;
+      boolean isPanic = result instanceof AbstractTruffleException && !(result instanceof DataflowError);
```

Comment:
Is there any system in these checks at all (another example at https://github.com/enso-org/enso/pull/7205/files#diff-bfe95bc1ca4b016dccbab4fc97f23809a6c83d040de6ca8a6a615f02378e1a59R357)?

I am afraid we are just patching random parts of the Enso system without having a consistent classification of the objects and their roles. Everything seems to hold together just with some unit tests. At least we have them, but still...

Reply:
I don't see a connection between this and the mentioned PR other than that we deal with DataFlowError. Yes, it's patching (not so random) parts of the Enso system and so what is wrong with that?
Alternatively we shouldn't match on `AbstractTruffleException` at all and wrap it into something we can control. Is it a better solution? That's debatable.

Reply:
> what is wrong with that?

It is perfectly _pragmatic behavior_. I am a huge fan of [pragmatism](http://wiki.apidesign.org/wiki/Pragmatic). However sometimes I a masking myself - shouldn't we be more [rationalistic]http://wiki.apidesign.org/wiki/Rationalism) when designing a language?

Possibly more consistency than patching would make the language better...